src/Language/Jsonnet/Parser.hs

```diff
@@ -343,10 +342,11 @@ objectP = Fix <$> annotateLoc (braces (try objectComp <|> object)) <?> "object"
           <$> option False (symbol "+" $> True) <*> sepP
       value <- exprP
       pure $ EField {..}
-    keyP = ((, True) <$> brackets exprP) <|> ((, False) <$> (unquoted <|> stringP))
+    keyP = ((,True) <$> brackets exprP) <|> ((,False) <$> (identStringP <|> stringP))
```

Comment:
note here that we are losing information (the fact that it's a valid identifier and thus jsonnet allows it to be unquoted). That's the reason I suggested have a separate type for keys. Maybe we can add an additional flag for when it's unquoted so that the pretty-printer can use that?

Reply:
Seeing this now after my comments on the pretty printer, perhaps is not worth to change anything here, so resolved!